### PR TITLE
Archive #8837

### PR DESCRIPTION
**Pull request by vbradnitski**

### CODE REVIEW

File: modules/core/core-content/src/main/java/com/enonic/xp/core/impl/content/ParentContentSynchronizer.java [hunk 1]
```diff
@@ -46,220 +50,269 @@ public ParentContentSynchronizer( @Reference final ContentService contentService
...
 
-                    if ( content.hasChildren() )
+            if ( result.getContents().isNotEmpty() )
+            {
+                final List<ContentToSync> childrenToSync = result.getContents().stream().map( content -> {
+                    final Context actualTargetContext = getActualContext( content.getId(), targetContexts.values() );
+
+                    return ContentToSync.create()
+                        .sourceContent( content )
+                        .targetContent( actualTargetContext != null
```
Comment (rymsha) [inline modules/core/core-content/src/main/java/com/enonic/xp/core/impl/content/ParentContentSynchronizer.java:228]: when actualTargetContent is null?

**Review by rymsha — CHANGES_REQUESTED** [review]

File: modules/core/core-api/src/main/java/com/enonic/xp/content/ContentConstants.java [hunk 2]
```diff
@@ -89,14 +62,29 @@
...
 
     public static final String UNPUBLISH_COMMIT_PREFIX = "COM_ENONIC_XP_CONTENT_UNPUBLISH";
 
+    public static final String RESTORE_COMMIT_PREFIX = "COM_ENONIC_XP_CONTENT_RESTORE";
+
+    public static final String ARCHIVE_COMMIT_PREFIX = "COM_ENONIC_XP_CONTENT_ARCHIVE";
+
     public static final String PUBLISH_COMMIT_PREFIX_DELIMITER = " ";
 
+    private static final PrincipalKey CONTENT_SUPER_USER_KEY = PrincipalKey.ofUser( IdProviderKey.system(), "content-su" );
```
[the remark this answers is not in the document]
Comment (vbradnitski) [reply modules/core/core-api/src/main/java/com/enonic/xp/content/ContentConstants.java:78]: I've no idea, this code is quite old. Maybe for better logs.

File: modules/core/core-api/src/main/java/com/enonic/xp/content/ContentVersionPublishInfo.java [hunk 3]
```diff
@@ -13,6 +13,8 @@
 
     private final String message;
 
+    private final CommitType type;
```
[the remark this answers is not in the document]
Comment (vbradnitski) [reply modules/core/core-api/src/main/java/com/enonic/xp/content/ContentVersionPublishInfo.java:16]: They do. It's actually a prefix in 'message' field

File: modules/core/core-content/src/main/java/com/enonic/xp/core/impl/content/ParentContentSynchronizer.java (hunk 1 above)
Comment (vbradnitski) [reply modules/core/core-content/src/main/java/com/enonic/xp/core/impl/content/ParentContentSynchronizer.java:228]: It's null if we sync new content in a project (source).

File: modules/core/core-content/src/main/java/com/enonic/xp/core/impl/content/MovedEventSyncCommand.java [hunk 4]
```diff
@@ -25,44 +32,99 @@ public static Builder create()
...
-                }
-            } );
-        }
+                } );
+            }
+        } );
     }
 
-    private boolean isToSyncParent( final Content targetContent )
+//    private void move()
```
Comment (rymsha) [inline modules/core/core-content/src/main/java/com/enonic/xp/core/impl/content/MovedEventSyncCommand.java:122]: remove

File: modules/core/core-content/src/main/java/com/enonic/xp/core/impl/content/ProjectContentEventListener.java [hunk 5]
```diff
@@ -92,60 +97,81 @@ private boolean isAllowedContentEvent( final String type )
...
+
+            final List<ProjectName> projectNames = nodes.stream()
+                .map( map -> map.get( "repo" ) )
+                .distinct()
+                .map( repo -> ProjectName.from( RepositoryId.from( repo ) ) )
+                .collect( Collectors.toList() );
+
+            if ( projectNames.size() != 1 )
+            {
+                throw new IllegalArgumentException( "An event cannot contain nodes from different repositories" );
```
Comment (rymsha) [inline modules/core/core-content/src/main/java/com/enonic/xp/core/impl/content/ProjectContentEventListener.java:123]: If projectNames is empty, error message will be misleading

File: modules/core/core-repo/src/main/java/com/enonic/xp/repo/impl/node/NodeServiceImpl.java [hunk 6]
```diff
@@ -632,14 +633,14 @@ public Node move( final MoveNodeParams params )
...
 
-        if ( moveNodeResult.getTargetNode() != null )
+        if ( !moveNodeResult.getMovedNodes().isEmpty() )
         {
-            this.eventPublisher.publish( NodeEvents.moved( moveNodeResult.getSourceNode(), moveNodeResult.getTargetNode() ) );
-            return moveNodeResult.getTargetNode();
+            this.eventPublisher.publish( NodeEvents.moved( moveNodeResult ) );
+            return moveNodeResult.getMovedNodes().get( 0 ).getNode();
         }
         else
```
Comment (rymsha) [inline modules/core/core-repo/src/main/java/com/enonic/xp/repo/impl/node/NodeServiceImpl.java:641]: Is this case necessary? If so, it should be tested

Comment (vbradnitski) [reply modules/core/core-repo/src/main/java/com/enonic/xp/repo/impl/node/NodeServiceImpl.java:641]: tested

File: modules/core/core-content/src/main/java/com/enonic/xp/core/impl/content/ParentContentSynchronizer.java (hunk 1 above)
Comment (vbradnitski) [reply modules/core/core-content/src/main/java/com/enonic/xp/core/impl/content/ParentContentSynchronizer.java:231]: tested

**Review by rymsha — APPROVED** [review]

**Review by anatol-sialitski — APPROVED** [review]

**Pull request merged**